static/js/pm_list.js

```diff
@@ -1,52 +1,92 @@
 import $ from "jquery";
+import _ from "lodash";
+
+import render_more_private_conversations from "../templates/more_pms.hbs";
 
-import * as narrow_state from "./narrow_state";
-import * as people from "./people";
 import * as pm_list_data from "./pm_list_data";
 import * as pm_list_dom from "./pm_list_dom";
+import * as resize from "./resize";
 import * as stream_popover from "./stream_popover";
 import * as ui from "./ui";
 import * as ui_util from "./ui_util";
 import * as vdom from "./vdom";
 
 let prior_dom;
-let private_messages_open = false;
 
 // This module manages the "Private messages" section in the upper
 // left corner of the app.  This was split out from stream_list.js.
 
-function get_filter_li() {
-    return $(".top_left_private_messages .private_messages_header");
+let private_messages_collapsed = false;
+
+// The private messages section can be zoomed in to view more messages.
+// This keeps track of if we're zoomed in or not.
+let zoomed = false;
+
+function get_private_messages_section_header() {
+    return $("#private_messages_section #private_messages_list .private-messages-section-header");
 }
 
-function set_count(count) {
-    ui_util.update_unread_count_in_dom(get_filter_li(), count);
+export function set_count(count) {
+    ui_util.update_unread_count_in_dom(get_private_messages_section_header(), count);
 }
 
 function remove_expanded_private_messages() {
     stream_popover.hide_topic_popover();
-    ui.get_content_element($("#private-container")).empty();
+    // While making the private messages list empty
+    // we check for any active conversation thread present in it and
+    // if yes, then append it below the collapsed PM section.
+    const $active_conversation_thread = $(".expanded_private_messages li.active-sub-filter");
+    $(".expanded_private_messages").empty().append($active_conversation_thread);
 }
 
-export function close() {
-    private_messages_open = false;
+function close() {
+    private_messages_collapsed = true;
     prior_dom = undefined;
     remove_expanded_private_messages();
 }
 
+function more_private_conversations_li(more_conversations_unread_count) {
+    // we render the li-item of `more conversations` with the unreads count inside the dom.
+    const render = () => render_more_private_conversations({more_conversations_unread_count});
+
+    // Used in vdom.js to check if an element has changed and should be updated in the DOM.
+    const eq = (other) =>
+        other.more_items &&
+        more_conversations_unread_count === other.more_conversations_unread_count;
+
+    const key = "more";
+
+    return {
+        key,
+        more_items: true,
+        more_conversations_unread_count,
+        render,
+        eq,
+    };
+}
```

Comment:
This function belongs in `pm_list_dom.js`, next to the parallel function.

Reply:
I've pushed an update to make this change.

Reply:
Cool, Thanks!